.github/workflows/pr_and_main.yml

```diff
@@ -0,0 +1,86 @@
+name: PR and main CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true  # cancel jobs from previous push
+
+jobs:
+  formatting:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - run: sudo apt-get install --yes clang-format-10
+      - run: find lib -type f -regex '.*\.\(cpp\|h\|tcc\)' -exec clang-format-10 -i {} +
+      - run: pip install cmake_format==0.6.9 flake8 yapf==0.30.0
+      - run: find . -type f -name CMakeLists.txt -or -name '*.cmake' | xargs cmake-format -i
+      - run: yapf --recursive --in-place .
+      - run: flake8 .
+      - run: |
+          find . -type f -not -path '.git/*' -exec sed -ri "s/[0-9]{4} (Scipp contributors)/$(date +%Y) \1/g" {} +
+          git diff --exit-code
+      - uses: stefanzweifel/git-auto-commit-action@v4
+
+  build_and_test:
+    needs: formatting
+    name: CI (${{ matrix.os }}, ${{ matrix.python-version }})
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash -l {0}  # required for conda env
+    env:
+      PYTHONPATH: ${{ github.workspace }}/install
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        python-version: ["3.7"]
```

Comment:
Do we stick with a single version for PRs?

Reply:
Yes, unchanged.